バグ改修・機能追加演習

wakanaiimura1203 opened the pull request:
(2-1)中級 ログイン者名表示
(3-1)初級 従業員一覧の並び順
(4-1)初級 日付フォーマット
(4-2)初級 価格フォーマット　
(4-3)中級 パンくずリスト修正
(6-1)中級 実務的エラー処理

wakanaiimura1203 committed:
(2-1)中級 ログイン者名表示
ログイン後、従業員一覧画面と従業員詳細画面右上の「〇〇さん　こんにちは！」に
ログインした人の名前を表示させる。

wakanaiimura1203 committed:
(3-1)初級 従業員一覧の並び順

wakanaiimura1203 committed:
(4-1)、(4-2)、(4-3)初級 日付・価格・パンくずリストの修正
・日付⇒入社日の表記を「2015年05月01日」に変更
・価格⇒給与を1000円単位でカンマ区切り表記に変更
・パンくず⇒従業員一覧へのリンクを記述

wakanaiimura1203 committed:
(6-1)中級 実務的エラー処理
エラー画面表示の為以下２ページを作成
・404.html
・500.html

hiroyukikanomata commented on src/main/resources/templates/error/500.html:
```diff
@@ -0,0 +1,11 @@
+</head>
+<body>
+ただいま緊急メンテナンスを行っております。<br>
+メンテナンス完了は11月18日 24:00を予定しております。<br>
```
楽しみにしてます。

hiroyukikanomata commented on src/main/resources/templates/error/500.html:
```diff
@@ -0,0 +1,11 @@
+<html lang="ja" xmlns:th="http://www.thymeleaf.org">
+<head>
+<meta charset="UTF-8">
+<title>500 Error Page</title>
```
タイトルで500エラーをばらさなくてもいいかも。直すほどでも無いけど。
ユーザに500エラーを知られちゃうと、信頼が下がるので。

hiroyukikanomata commented on src/main/resources/templates/employee/detail.html:
```diff
@@ -101,7 +101,7 @@
 							    </th>
 							    <td>
-							      <span th:utext="${employee.hireDate}">2012/11/29</span>
+							      <span th:text="${#dates.format(employee.hireDate,'yyyy年MM月dd日')}">2012/11/29</span>
```
ここはいいんだけど、list.html内にも入社日表示部分ありませんでした？

hiroyukikanomata commented on src/main/java/jp/co/sample/emp_management/repository/EmployeeRepository.java:
```diff
@@ -50,7 +50,7 @@
 	 */
 	public List<Employee> findAll() {
-		String sql = "SELECT id,name,image,gender,hire_date,mail_address,zip_code,address,telephone,salary,characteristics,dependents_count FROM employees";
+		String sql = "SELECT id,name,image,gender,hire_date,mail_address,zip_code,address,telephone,salary,characteristics,dependents_count FROM employees ORDER BY hire_date";
```
古株の社員から表示している？

hiroyukikanomata reviewed (changes requested):
一部直すといいです。

wakanaiimura1203 committed:
以下３点修正済み。
・【500.html】⇒タイトルを「メンテナンス中」に修正
・【list.html】⇒入社日表示部分を修正
・【EmployeeRepository】⇒入社日が最新の人から順に表示に変更
